src/rules/pydocstyle/rules/multi_line_summary_start.rs

```diff
@@ -27,21 +30,57 @@ pub fn multi_line_summary_start(checker: &mut Checker, docstring: &Docstring) {
             .rules
             .enabled(&Rule::MultiLineSummaryFirstLine)
         {
-            checker.diagnostics.push(Diagnostic::new(
+            let mut diagnostic = Diagnostic::new(
                 violations::MultiLineSummaryFirstLine,
                 Range::from_located(docstring.expr),
-            ));
+            );
+            if checker.patch(diagnostic.kind.rule()) {
+                let loc = docstring.expr.location;
+                let mut endl = loc.row() + 1;
+                // Delete until first non-whitespace char.
+                for line in content_lines {
+                    if let Some(endc) = line.find(|c: char| !c.is_whitespace()) {
+                        let start = Location::new(loc.row(), loc.column() + first_line.len());
+                        let end = Location::new(endl, endc);
+                        diagnostic.amend(Fix::deletion(start, end));
+                        break;
+                    }
+                    endl += 1;
+                }
+            }
+            checker.diagnostics.push(diagnostic);
         }
     } else {
         if checker
             .settings
             .rules
             .enabled(&Rule::MultiLineSummarySecondLine)
         {
-            checker.diagnostics.push(Diagnostic::new(
+            let mut diagnostic = Diagnostic::new(
                 violations::MultiLineSummarySecondLine,
                 Range::from_located(docstring.expr),
-            ));
+            );
+            if checker.patch(diagnostic.kind.rule()) &&
+                // Skip cases where leading quote is preceded by non-whitespace for now,
+                // e.g. `def foo(): """...`. Figuring out correct indentation after the newline
+                // we'd add would require some work.
+                docstring.indentation.chars().all(char::is_whitespace)
```

Comment:
This is to protect against breakage with https://github.com/charliermarsh/ruff/blob/ebfdefd110e83bcd1d101ee5a424fc6a82e020b5/resources/test/fixtures/pydocstyle/D.py#L438
The docstring's `indentation` ends up as `def docstring_start_in_same_line(): ` which we obviously cannot insert as indentation after the newline we'd insert. I guess it's fine to just skip these cases for now.

If that's acceptable, is this an ok way to conditionalize the fix, or is there some other preferred way? The thing is marked as an `AlwaysAutofixableViolation` but with this, it would not be _always_ autofixable. The CLI output says "potentially fixable" for these, and there doesn't seem to be a concept of codified conditional fixability, so I guess it's ok.

Reply:
A few things:

1. We do have a pattern for a rule that's conditionally fixable. If you grep for `autofix_title_formatter`, you'll see some examples of that :)
2. We probably _could_ fix those cases, or at least try, although I'd still merge this even if we don't. `checker.stylist.indentation()` will contain the inferred indentation for the file... so you could look at the current indentation of the `def`, then add `checker.stylist.indentation()` to it. That could be wrong in some cases, like files with mixed indentation, but _should_ otherwise be correct. I'll leave it to you whether you want to enable that behavior :)

Reply:
Thanks for the advice! I opted to go for 2), mostly for practice purposes, 6a7f2f15d2713106f4ee015f71e886977b35ee5b has it. There are still some cases in which we skip the fix (parent definition is not function/class, or their indentation contains something besides whitespace), but I'm not sure if they can actually happen, or are more on the theoretical side.

The implementation reads a bit clumsy to me now, but that's what my Rust-fu is up to at time of writing :|